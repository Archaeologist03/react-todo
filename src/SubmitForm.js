import React from 'react';

function SubmitForm(props) {


    return (
        <div className="submit">
            <input 
                type="text" 
                autoFocus={true}
                onChange={props.typing} 
                value={props.inputVal}/>
            <button 
                className="submitBtn" 
                type="submit" 
                onClick={props.click}>
                    Add
            </button>
        </div>
    )
}

export default SubmitForm;