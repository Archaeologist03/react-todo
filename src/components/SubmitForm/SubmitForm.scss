.submit-container {
  height: 100px;
  width: 100%;
  border: 2px solid burlywood;
  padding-left: 30px;
  padding-top: 20px;

  .addItemInput {
    border-radius: 5px;
    padding: 5px 0;
    margin-right: 10px;
    background: #da4453;
    background: -webkit-linear-gradient(to right, #89216b, #da4453);
    background: linear-gradient(to right, #89216b57, #da445357);
    background: rgba(248, 82, 21, 0.897);
    color: whitesmoke;
    font-weight: bold;
    box-shadow: 0px 4px 10px #d1bd09b0;
    text-shadow: 0px 1px 2px;
    font-size: 16px;

    &::placeholder {
      color: rgba(247, 239, 229, 0.747);
    }
  }

  .submitBtn {
    /* padding: 10px; */
    width: 50px;
    height: 50px;
    border-radius: 50%;
    outline: none;
    background-color: rgba(248, 82, 21, 0.897);
    color: rgb(3, 12, 20);
    border: none;
    transition: 1s ease-out;
    transform: rotateZ(0);

    &:hover {
      background-color: rgba(240, 69, 6, 0.897);
      transform: rotateZ(360deg);
    }
    &:active {
      transform: scale(1.15);
      transition: 0.25s;
    }

    &:hover .addBtnText {
      color: rgb(126, 5, 1);
      animation: btnTextMove 1s cubic-bezier(0.25, 1.04, 0.65, 0.07);
    }

    .addBtnText {
      display: block;
      transition: 0.5s ease-out;
    }
  }
}
