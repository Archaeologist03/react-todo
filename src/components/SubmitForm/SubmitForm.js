import React from 'react';
import { connect } from 'react-redux';

import { inputChange, updateInitialState } from '../../store/actions/app';
import { addToList } from '../../store/actions/listActions';

import './SubmitForm.scss';

function SubmitForm(props) {
  const handleInputEnter = e => {
    if (e.key === 'Enter') {
      props.onAddToList(props.text);
    }
  };

  const handlerBtnSubmit = e => {
    e.preventDefault();
    // this prevents empty string to go to server, therefore we lose err message, but cut extra complexity on AC and server.
    if (props.text) {
      props.onAddToList(props.text);
    }
  };

  return (
    <div className='submit-container'>
      <input
        className='addItemInput'
        type='text'
        autoFocus={true}
        onChange={props.onInputChange}
        value={props.text}
        onKeyPress={handleInputEnter}
        placeholder='Add task'
      />
      <button className='submitBtn' type='submit' onClick={handlerBtnSubmit}>
        <span className='addBtnText'>Add</span>
      </button>
    </div>
  );
}

const mapStateToProps = state => {
  return {
    text: state.app.text,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    updateInitialState: () => dispatch(updateInitialState()),
    onInputChange: text => dispatch(inputChange(text)),
    onAddToList: newItem => dispatch(addToList(newItem)),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SubmitForm);
