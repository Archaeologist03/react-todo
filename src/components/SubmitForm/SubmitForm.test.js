import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { SubmitForm } from './SubmitForm';

describe('SubmitForm', () => {
  it('should render SubmitForm without crashing given the required props', () => {
    const props = {
      text: '',
      updateInitialState: jest.fn(),
      onInputChange: jest.fn(),
      onAddToList: jest.fn(),
    };
    const wrapper = shallow(<SubmitForm {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call onAddToList fn on click with text arg', () => {
    const props = {
      text: '',
      updateInitialState: jest.fn(),
      onInputChange: jest.fn(),
      onAddToList: jest.fn(),
    };

    const wrapper = shallow(<SubmitForm {...props} />);
    const submitBtnClassElement = wrapper.find('.submitBtn');

    submitBtnClassElement.simulate('click', {
      preventDefault: () => {
        props.onAddToList(props.text);
      },
    });
    expect(props.onAddToList).toBeCalledWith(props.text);
  });

  it('should call onAddToList fn on Enter keypress with text arg', () => {
    const props = {
      text: '',
      updateInitialState: jest.fn(),
      onInputChange: jest.fn(),
      onAddToList: jest.fn(),
    };

    const wrapper = shallow(<SubmitForm {...props} />);
    const addItemInputClassElement = wrapper.find('.addItemInput');

    addItemInputClassElement.simulate('keypress', { key: 'Enter' });
    expect(props.onAddToList).toBeCalledWith(props.text);
  });

  it('should call onInputChange fn on onChange event with text arg', () => {
    const props = {
      text: 'task1',
      updateInitialState: jest.fn(),
      onInputChange: jest.fn(),
      onAddToList: jest.fn(),
    };

    const wrapper = shallow(<SubmitForm {...props} />);
    const addItemInputClassElement = wrapper.find('.addItemInput');

    addItemInputClassElement.simulate('change', props.text);
    expect(props.onInputChange).toBeCalledWith(props.text);
  });
});
