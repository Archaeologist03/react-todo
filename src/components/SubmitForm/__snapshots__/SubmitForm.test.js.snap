// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitForm should render SubmitForm without crashing given the required props 1`] = `
<div
  className="submit-container"
>
  <input
    autoFocus={true}
    className="addItemInput"
    onChange={[MockFunction]}
    onKeyPress={[Function]}
    placeholder="Add task"
    type="text"
    value=""
  />
  <button
    className="submitBtn"
    onClick={[Function]}
    type="submit"
  >
    <span
      className="addBtnText"
    >
      Add
    </span>
  </button>
</div>
`;
